v3/how-to-guides/04-ocw/b-http-requests/index.mdx

```diff
@@ -0,0 +1,140 @@
+---
+title: Making HTTP Requests
+slug: /how-to-guides/v3/ocw/http-requests
+keywords: ['off-chain worker', 'ocw', 'http', 'https', 'requests']
+version: '3.0'
+section: how to guides
+category: ocw
+---
+
+Making HTTP Requests
+
+<Objectives data={[
+  {
+    title: 'Goal',
+    description: `Send http requests offchain`
+  },
+  {
+    title: 'Use Cases',
+    description: `GET data from offchain, or POST data offchain.`,
+  },
+]}/>
+
+## Overview
+
+The offchain-worker (**ocw** for short)  was introduced to simplify the building of oracles. Because a blockchain does not have access to data outside its own network, an oracle is required to provide context on decision Making
+
+In this chapter we will look through retreiving/submitting data via an API. 
+
+Remember that although Rust provides various libraries that allow you to issue HTTP requests, the OCW runs in an [no-std](https://docs.rust-embedded.org/book/intro/no-std.html) environment, luckily Substrate provides us a few libraries we can use to issue HTTP requests to an API. 
+
+### Supported Methods
+The Substrate HTTP library supports the following methods:
+- GET
+- POST
+- PUT
+- PATCH
+- DELETE
+
+
+## Get Request
+
+1. Define a request block
+  ```rust
+        // We want to keep the offchain worker execution time reasonable, so we set a hard-coded
+		// deadline to 2s to complete the external call.
+		// You can also wait idefinitely for the response, however you may still get a timeout
+		// coming from the host machine.
+		let deadline = sp_io::offchain::timestamp().add(Duration::from_millis(2_000));
+
+        // Initiate an external HTTP GET request.
+		// This is using high-level wrappers from `sp_runtime`, for the low-level calls that
+		// you can find in `sp_io`. The API is trying to be similar to `reqwest`, but
+		// since we are running in a custom WASM execution environment we can't simply
+		// import the library here.
+		let request = http::Request::get("https://min-api.cryptocompare.com/data/price?fsym=BTC&tsyms=USD");
+		// We set the deadline for sending of the request, note that awaiting response can
+		// have a separate deadline. Next we send the request, before that it's also possible
+		// to alter request headers or stream body content in case of non-GET requests.
```

Comment:
Not sure what you're trying to say here / how it corresponds to line 59.

Reply:
I got the comments off the ocw example on substrate https://github.com/paritytech/substrate/blob/monthly-2021-12/frame/examples/offchain-worker/src/lib.rs#L582-#L584

I think it just means that for some APIs there are certain headers that need to be added for each request, but I can edit it to make more sense